### PR TITLE
Document Rally use with Elastic Serverless

### PR DESCRIPTION
**Pull request by gbanasiak**

### CODE REVIEW

File: docs/serverless.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Rally
+-----
+
+Rally determines if it is benchmarking a Serverless cluster looking at ``version.build_flavor`` attribute returned by Elasticsearch root endpoint (``/``). If Serverless is confirmed, Rally *automatically* skips all tasks based on :ref:`operations<track_operations>` status in Serverless. For instance, a task with :ref:`node-stats<operation_node_stats>` operation is skipped because ``/_nodes/stats`` Elasticsearch API is not available in Serverless. The status determination is hardcoded into Rally, so if the API status changes in Elasticsearch, Rally needs to be updated too. The same skip logic applies to :doc:`/telemetry`.
+
+.. note::
+
+    Elasticsearch Serverless API is not publicly documented yet. Today, the ultimate documentation is Elasticsearch `source code <https://github.com/elastic/elasticsearch>`_. Please look for ``@ServerlessScope(Scope.PUBLIC)`` annotation to determine if a given REST action is available in Serverless.
```
Comment (inqueue) [inline docs/serverless.rst:22]:
```suggestion
    The Elasticsearch Serverless API is not yet publicly documented. Today, the ultimate documentation is Elasticsearch `source code <https://github.com/elastic/elasticsearch>`_. Please look for the ``@ServerlessScope(Scope.PUBLIC)`` annotation to determine if a given REST action is available in Serverless.
```

File: docs/serverless.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+The automatic task skip can be overridden through :ref:`run-on-serverless<track_schedule>` property::
+
+  {
+    [..]
+    "schedule": [
+      {
+        "operation": "node-stats",
+        "clients": 1,
+        "run-on-serverless": True
```
Comment (inqueue) [inline docs/serverless.rst:32]: Using a JSON boolean value is usually preferred over Python, although Rally might already deal with it appropriately:
```suggestion
        "run-on-serverless": true
```

File: docs/serverless.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+If there is a need to skip or modify behaviour of any of the above when benchmarking Serverless cluster, this can be achieved through modification of a track definition (see next section).
+
+Furthermore, some operations behave differently when used in the Serverless context. Specifically:
+
+- ``delete-composable-templates`` ignores ``delete-matching-indices`` property,
+- ``cache`` property in all search operations (``search``, ``paginated-search``, ``scroll-search``, ``composite-agg``) defaults to ``false``.
+
+Tracks
+------
+Track files use Jinja2 as a template language with a few :ref:`extensions<advanced_extensions>`. To accommodate for Serverless, ``build_flavor`` global variable was added. It resolves to ``serverless`` when benchmarking a Serverless cluster. The primary use case for this variable is conditional exclusion of index settings which are unavailable or hidden from users in Serverless.
```
Comment (inqueue) [inline docs/serverless.rst:63]:
```suggestion
Track files use Jinja2 as a template language with a few :ref:`extensions<advanced_extensions>`. To accommodate for Serverless, the ``build_flavor`` global variable was added. It resolves to ``serverless`` when benchmarking a Serverless cluster. The primary use case for this variable is conditional exclusion of index settings that are unavailable or hidden from users in Serverless.
```

File: docs/serverless.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+.. note::
+
+    The index setting status in Serverless is not publicly documented yet. Today, the ultimate documentation is Elasticsearch `source code <https://github.com/elastic/elasticsearch>`_. Please look for ``Property.ServerlessPublic`` property to determine if a given index setting is publicly available in Serverless.
```
Comment (inqueue) [inline docs/serverless.rst:67]:
```suggestion
    The index setting status in Serverless is not publicly documented yet. Today, the ultimate documentation is Elasticsearch `source code <https://github.com/elastic/elasticsearch>`_. Please look for the ``Property.ServerlessPublic`` property to determine if a given index setting is publicly available in Serverless.
```

File: docs/track.rst [hunk 5]
```diff
@@ -1893,7 +1895,7 @@ If you want it to delete one specific index template instead, you can specify th
 
 * ``template`` (mandatory): The name of the composable template that should be deleted.
 * ``only-if-exists`` (optional, defaults to ``true``): Defines whether the index template should only be deleted if it exists.
-* ``delete-matching-indices`` (optional, defaults to ``false``): Whether to delete indices that match the index template's index pattern.
+* ``delete-matching-indices`` (optional, defaults to ``false``): Whether to delete indices that match the index template's index pattern. This setting is ignored in `Elastic Serverless <https://docs.elastic.co/serverless>`_ - please use data streams and ``delete-data-stream`` operation instead.
```
Comment (inqueue) [inline docs/track.rst:1898]:
```suggestion
* ``delete-matching-indices`` (optional, defaults to ``false``): Whether to delete indices that match the index template's index pattern. This setting is ignored in `Elastic Serverless <https://docs.elastic.co/serverless>`_ - please use data streams and the ``delete-data-stream`` operation instead.
```

File: docs/track.rst [hunk 6]
```diff
@@ -1893,7 +1895,7 @@ If you want it to delete one specific index template instead, you can specify th
...
 * ``index-pattern`` (mandatory iff ``delete-matching-indices`` is ``true``): Specifies the index pattern to delete.
```
Comment (inqueue) [inline docs/track.rst:1899]: Noticed while passing through:
```suggestion
* ``index-pattern`` (mandatory if ``delete-matching-indices`` is ``true``): Specifies the index pattern to delete.
```

**Review by inqueue — APPROVED** [review]
I left a few comments- LGTM overall. Thanks!

File: docs/track.rst (hunk 6 above)
Comment (gbanasiak) [reply docs/track.rst:1899]: The "iff" is an acronym for "if and only if". It's present in a couple of places in Rally code and docs. I'll remove its occurrences in docs. The first time I saw it I also thought it's a typo.

**Comment by gbanasiak** [comment]
@inqueue Many thanks for the review and all the improvements!

File: docs/index.rst [hunk 7]
```diff
@@ -7,7 +7,7 @@ Rally |release|
     .. warning::
 
         This documentation is for the version of Rally currently under development.
-        Were you looking for the `documentation of the latest stable version <//esrally.readthedocs.io/en/stable/>`_?
+        Were you looking for the `documentation of the latest stable version <https://esrally.readthedocs.io/en/stable/>`_?
```
Comment (pquentin) [inline docs/index.rst:10]: In case you were wondering, this is a protocol-relative URL that was useful when Read the Docs wasn't enforcing HTTPS.

**Pull request merged**